notebook/notebookapp.py

```diff
@@ -550,6 +555,10 @@ def _default_log_format(self):
         help=_("Whether to allow the user to run the notebook as root.")
     )
 
+    allow_hidden = Bool(False, config=True,
```

Comment:
It seems like this would more logically be on `FileContentsManager`, rather than NotebookApp. What do you think?

Reply:
Interestingly, the abstract ContentsManager (which doesn't know about files) defines an `is_hidden()` method, but doesn't appear to use it for anything. It's FileContentsManager which is deciding not to show hidden files.

Reply:
Hrm I think, for my original intentions, it would be mildly confusing to put it in `FileContentsManager`.

* `FileContentsManager` controls the ability to travel to hidden directories.
* `AuthenticatedFileHandler` controls the ability to access hidden files in markdown (maybe more?)
* `TreeHandler` controls the ability to access hidden files in the filebrowser

For my personal use, this PR was to allow access to hidden files through markdown, not to allow travel to hidden directories/open hidden files. But issue #2659 wanted that ability, so I'm not sure what the best course of action is.

If you think I should include the ability to travel to hidden directories and access hidden files, I can though. Then each class will have their own `allow_hidden` and the option can be accessed separately.  Although `FileContentsManager` and `TreeHandler` should probably be linked somehow (can't think of a reason to separate that logic.)